nsv13/code/game/gamemodes/overmap/objectives/scan.dm

```diff
@@ -0,0 +1,43 @@
+/datum/overmap_objective/scan
+	name = "Scan anomalies"
+	desc = "Scan a target anomaly_type"
+	brief = "Locate and scan target anomaly_type"
+	var/anomaly_type
+	var/anomaly_name = "anomaly"
+	var/list/anomaly_whitelist = list(
+		/obj/effect/overmap_anomaly/safe/sun,
+		/obj/effect/overmap_anomaly/safe/sun/red_giant,
+		/obj/effect/overmap_anomaly/wormhole,
+		/obj/effect/overmap_anomaly/singularity)
+	var/minimum_anomalies = 1
+	var/maximum_anomalies = 5
+
+/datum/overmap_objective/scan/New()
+	. = ..()
+	var/list/valid_anomalies = anomaly_whitelist
+	target = rand(minimum_anomalies, maximum_anomalies)
+	while(length(valid_anomalies) && !anomaly_type) //doing this to check if there are enough of this type
+		anomaly_type = pick(valid_anomalies)
+		anomaly_name = get_atom_by_type(GLOB.overmap_anomalies, anomaly_type)?.name
+		if(!anomaly_name || (count_by_type(GLOB.overmap_anomalies, anomaly_type) < target))
+			valid_anomalies -= anomaly_type
+			anomaly_type = null
+			anomaly_name = "ERROR"
+	if(!anomaly_type)
+		message_admins("Overmap gamemode failed to locate enough anomalies! Setting to auto-complete.")
+		status = 1
```

Comment:
```suggestion
	while(length(valid_anomalies) && !anomaly_type) //doing this to check if there are enough of this type
		anomaly_type = pick(valid_anomalies)
		if(count_by_type(GLOB.overmap_anomalies, anomaly_type) < target)
			valid_anomalies -= anomaly_type
			anomaly_type = null
			anomaly_name = "ERROR"
		else
			anomaly_name = initial(anomaly_type.name)
	if(!anomaly_type)
		message_admins("Overmap gamemode failed to locate enough anomalies! Setting to auto-complete.")
		status = 1
```

Reply:
initial(anomaly_type.name) won't work because you can't access variables of a typepath

Reply:
You actually can, that's how (base ss13) game modes read their variables without instantiating them all

Reply:
Huh, it just threw a runtime when I tried to do it

Reply:
Yeah looking at gamemodes there's a comment stating they DO instance it because they can't find another way either (line 248 of controllers/configuration/configuration.dm)